imageregistry/v1/types.go

```diff
@@ -306,6 +306,81 @@ type ImageRegistryConfigStorageIBMCOS struct {
 	ServiceInstanceCRN string `json:"serviceInstanceCRN,omitempty"`
 }
 
+// EndpointAccessibility defines the Alibaba VPC endpoint for storage
+type EndpointAccessibility string
+
+// AlibabaEncryptionMode defines an enumerable type for the encryption mode
+type AlibabaEncryptionMode string
+
+const (
+	// InternalEndpoint sets the VPC endpoint to internal
+	InternalEndpoint EndpointAccessibility = "Internal"
+	// PublicEndpoint sets the VPC endpoint to public
+	PublicEndpoint EndpointAccessibility = "Public"
+
+	// PlainText is an AlibabaEncryptionMode. This is the dafault. This means no encryption
+	PlainText AlibabaEncryptionMode = "PlainText"
+	// AES256 is an AlibabaEncryptionMode. This means AES256 encryption
+	AES256 AlibabaEncryptionMode = "AES256"
+	// KMS is an AlibabaEncryptionMode. This means KMS encryption
+	KMS AlibabaEncryptionMode = "KMS"
+)
+
+// EncryptionAlibaba this a union type in kube parlance.  Depending on the value for the encryptionType,
+// different pointers may be used
+type EncryptionAlibaba struct {
+	// EncryptionType defines the different encrytion types available
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default is `PlainText`.
+	// +kubebuilder:validation:Enum="PlainText";"KMS";"AES256"
+	// +kubebuilder:default="PlainText"
+	// +kubebuilder:validation:Required
```

Comment:
with the default this must be optional now because client-side validation does not see the default.

Reply:
Updated to optional.